### DISCUSSION TITLE
Does it mean that shifting to java-client 8.5.1 from 8.5.0, will make the methods using the deprecated TouchAction NOT function any more ?

### DISCUSSION THREAD

**Discussion by vnuta9** [appium/java-client#1943] [General]
I have a bunch of methods in my code which are using tap, swipe actions by utilizing the deprecated class **io.appium.java_client.TouchAction.** By shifting to 8.5.1, will all of those methods won't even compile  ???

Thanks for your response.

**Comment by saikrishna321** [comment]
Not in 8.5.1. Moving forward these deprecated methods will be removed. Thats when you will have trouble. 

We recommend using Actions API for gestures.

**Comment by vnuta9** [comment]
Dear @saikrishna321 ;
Thanks for your reply.
and thanks for your great contribution to Appium Java-client development and support.
